File: library/src/main/java/com/afollestad/aesthetic/BottomNavBgMode.java
package com.afollestad.aesthetic;

/** @author Aidan Follestad (afollestad) */
public enum BottomNavBgMode {
  BLACK_WHITE_AUTO(0),
  PRIMARY(1),
  PRIMARY_DARK(2),
  ACCENT(3);

  private final int value;

  BottomNavBgMode(int value) {
    this.value = value;
  }

  public int toInt() {
    return value;
  }

  public static BottomNavBgMode fromInt(int value) {
    switch (value) {
      case 0:
        return BLACK_WHITE_AUTO;
      case 1:
        return PRIMARY;
      case 2:
        return PRIMARY_DARK;
      default:
        return ACCENT;
    }
  }
}

File: library/src/main/java/com/afollestad/aesthetic/BottomNavBgMode.kt
/*
 * Licensed under Apache-2.0
 *
 * Designed and developed by Aidan Follestad (@afollestad)
 */
package com.afollestad.aesthetic

/** @author Aidan Follestad (afollestad) */
enum class BottomNavBgMode(val value: Int) {
  BLACK_WHITE_AUTO(0),
  PRIMARY(1),
  PRIMARY_DARK(2),
  ACCENT(3);

  companion object {
    fun fromInt(value: Int): BottomNavBgMode {
      return when (value) {
        0 -> BLACK_WHITE_AUTO
        1 -> PRIMARY
        2 -> PRIMARY_DARK
        else -> ACCENT
      }
    }
  }
}
